plugins/nordic-hid/fu-nordic-hid-archive.c

```diff
@@ -11,15 +11,157 @@
 #include "fu-nordic-hid-firmware-b0.h"
 #include "fu-nordic-hid-firmware-mcuboot.h"
 
-/* current version format is 0 */
-#define MAX_VERSION_FORMAT 0
+/* The plugin currently supports version format of either 0 or 1. */
+#define MIN_VERSION_FORMAT 0
+#define MAX_VERSION_FORMAT 1
 
 struct _FuNordicHidArchive {
 	FuFirmwareClass parent_instance;
 };
 
 G_DEFINE_TYPE(FuNordicHidArchive, fu_nordic_hid_archive, FU_TYPE_FIRMWARE)
 
+static gboolean
+fu_nordic_hid_archive_parse_file_image_create(JsonObject *obj,
+					      FuFirmware **image,
+					      const gchar **bootloader_name,
+					      GError **error)
+{
+	if (json_object_has_member(obj, "version_B0")) {
+		*bootloader_name = "B0";
+		*image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_B0, NULL);
+	} else if (json_object_has_member(obj, "version_MCUBOOT")) {
+		*bootloader_name = "MCUBOOT";
+		*image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_MCUBOOT, NULL);
+	} else if (json_object_has_member(obj, "version_MCUBOOT+XIP")) {
+		*bootloader_name = "MCUBOOT+XIP";
+		*image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_MCUBOOT, NULL);
+	} else {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "only B0 and MCUboot bootloaders are supported");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
```

Comment:
can we return the board name as a `static gchar *`  and NULL on error?

Reply:
Aligned approach also for `flash_area_id` getter (to improve consistency)